tests/unit/src/main/java/org/mule/tck/util/MuleContextUtils.java

```diff
@@ -75,13 +90,76 @@ public static MuleContext mockMuleContext() {
+  public static MuleContextWithRegistries mockContextWithServices() {
+    final MuleContextWithRegistries muleContext = mockMuleContext();
     when(muleContext.getSchedulerService()).thenReturn(spy(new SimpleUnitTestSupportSchedulerService()));
     ErrorTypeRepository errorTypeRepository = mock(ErrorTypeRepository.class);
     when(muleContext.getErrorTypeRepository()).thenReturn(errorTypeRepository);
     when(errorTypeRepository.getErrorType(any(ComponentIdentifier.class))).thenReturn(empty());
     mockNotificationsHandling(muleContext.getRegistry());
+
+    // Ensure propagation of mock MuleContexts through
+    when(muleContext.getInjector()).thenReturn(new Injector() {
+
+      @Override
+      public <T> T inject(T object) throws MuleException {
+        for (Field field : getAllFields(object.getClass(), withAnnotation(Inject.class))) {
+          Class<?> dependencyType = field.getType();
+
+          boolean nullToOptional = false;
+          if (dependencyType.equals(Optional.class)) {
+            Type type = ((ParameterizedType) (field.getGenericType())).getActualTypeArguments()[0];
+            if (type instanceof ParameterizedType) {
+              dependencyType = (Class<?>) ((ParameterizedType) type).getRawType();
+            } else {
+              dependencyType = (Class<?>) type;
+            }
+            nullToOptional = true;
+          }
+
+          if (MuleContext.class.isAssignableFrom(dependencyType)) {
+            try {
+              field.setAccessible(true);
+              field.set(object, nullToOptional ? of(muleContext) : muleContext);
+            } catch (Exception e) {
+              throw new RuntimeException(format("Could not inject dependency on field %s of type %s", field.getName(),
+                                                object.getClass().getName()),
+                                         e);
+            }
+          }
+        }
+        for (Method method : getAllMethods(object.getClass(), withAnnotation(Inject.class))) {
+          if (method.getParameters().length == 1) {
+            Class<?> dependencyType = method.getParameterTypes()[0];
+
+            boolean nullToOptional = false;
+            if (dependencyType.equals(Optional.class)) {
+              Type type = ((ParameterizedType) (method.getGenericParameterTypes()[0])).getActualTypeArguments()[0];
+              if (type instanceof ParameterizedType) {
+                dependencyType = (Class<?>) ((ParameterizedType) type).getRawType();
+              } else {
+                dependencyType = (Class<?>) type;
+              }
+              nullToOptional = true;
+            }
+
+            if (MuleContext.class.isAssignableFrom(dependencyType)) {
+              try {
+                method.invoke(object, nullToOptional ? of(muleContext) : muleContext);
+              } catch (Exception e) {
+                throw new RuntimeException(format("Could not inject dependency on method %s of type %s", method.getName(),
```

Comment:
Extract exception creation into an utility method

Reply:
the messages are differnet